libs/go/sia/gcp/functions/gcf_get_sia_cert.go

```diff
@@ -0,0 +1,263 @@
+package functions
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"github.com/AthenZ/athenz/clients/go/zts"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// GcfGetSiaCerts this method can be called from within a GCF (Google Cloud Function) -
+//
+//	to get an Athenz certificate from ZTS.
+//
+// This file should usually be copied without changes into the GCF source-code.
+//
+// See https://cloud.google.com/functions/docs/writing/write-http-functions#http-example-go
+func GcfGetSiaCerts(
+	athenzDomain string,
+	athenzService string,
+	gcpProjectId string,
+	athenzProvider string,
+	ztsUrl string,
+	certDomain string,
+	optionalSubjectFields CsrSubjectFields,
+) (*SiaCertData, error) {
+
+	athenzDomain = strings.ToLower(athenzDomain)
+	athenzService = strings.ToLower(athenzService)
+	athenzProvider = strings.ToLower(athenzProvider)
+
+	// Get an identity-document for this GCF from GCP.
+	attestationData, err := getGcpFunctionAttestationData(ztsUrl)
+	if err != nil {
+		return nil, err
+	}
+	//log.Printf("GCP Attestation Data: %s", attestationData) // commented out - sensitive info
```

Comment:
let's remove all the commented log entries

Reply:
👍